extensions/flac/src/main/jni/flac_parser.cc

```diff
@@ -172,6 +172,30 @@ void FLACParser::metadataCallback(const FLAC__StreamMetadata *metadata) {
     case FLAC__METADATA_TYPE_SEEKTABLE:
       mSeekTable = &metadata->data.seek_table;
       break;
+    case FLAC__METADATA_TYPE_VORBIS_COMMENT:
+      if (!mVorbisCommentValid) {
+        FLAC__uint32 count = 0;
+        const FLAC__StreamMetadata_VorbisComment *vc =
+            &metadata->data.vorbis_comment;
+        mVorbisCommentValid = true;
+        mVorbisComment.metadataArray =
```

Comment:
These aren't freed anywhere.
Instead of malloc and copy, can we hold reference to &vc->comments[i]; and create required strings in flac_jni.cc flacDecodeVorbisComment()?
i.e. getVorbisComment() can return FLAC__StreamMetadata_VorbisComment* 
That would avoid the need for malloc and copy here.

Reply:
I have reorganised the code according to other comments, and added the free to the character arrays that will be used.